src/integration/java/com/ibm/eventstreams/connect/mqsource/MQSourceTaskAuthIT.java

```diff
@@ -139,6 +140,38 @@ public void verifyJmsConnClosed() throws Exception {
         stopper.run();
     }
 
+
+
+    @Test
+    public void verifyJmsConnections() throws Exception {
+
+        final int restApiPortNumber = mqContainer.getMappedPort(9443);
+        final MQSourceTask connectTask = new MQSourceTask();
+        connectTask.start(getConnectorProps());
+
+       // count number of connections to the qmgr at the start
+        final int numQmgrConnections = MQQueueManagerAttrs.getNumConnections(QMGR_NAME, restApiPortNumber,
+        ADMIN_PASSWORD);
+
+        final Map<String, String> connectorConfigProps = getConnectorProps();
+        connectorConfigProps.put("topic", "TOPICA");
+        connectTask.stop();
```

Comment:
This line should be removed. 

This is why the test was passing without your change - by stopping the Connector completely (instead of leaving it running but passing it new config) you called this code: 

https://github.com/ibm-messaging/kafka-connect-mq-source/blob/0119b06302d897609e925df0df091845330159a1/src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceTask.java#L237-L239

This caused the JMSReader to be explicitly closed and cleaned up. 

This is why this test passes, and why it doesn't recreate the problem that this pull request needs to address (specifically, that old connections are left open when a running Connector is given new config)